test/functional/tests/fault_injection/test_standby_activate_different_cache_path.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import pytest
+
+from api.cas import casadm
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_utils.filesystem.symlink import Symlink
+from test_utils.size import Size, Unit
+from api.cas.cache_config import CacheLineSize
+from api.cas.cli import standby_activate_cmd
+from api.cas.cache import CacheStatus
+from test_tools.dd import Dd
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+def test_standby_activate_different_cache_path(cache_line_size):
+    """
+    title: test_standby_activate_different_cache_path
+    description: |
+        Symlink-activated CAS partitions distinction test.
+    pass_criteria:
+      - Cache is successfully activated with symlink on second partition.
+      - Cache is present and running.
+    """
+    with TestRun.step("Prepare two partitions with the same size."):
+        cache_dev = TestRun.disks["cache"]
+        cache_dev.create_partitions([Size(200, Unit.MebiByte), Size(200, Unit.MebiByte)])
+        active_cache_dev = cache_dev.partitions[0]
+        standby_cache_dev = cache_dev.partitions[1]
+        cache_id = 1
+
+    with TestRun.step("Start a regular cache instance on one of the partitions and stop it."):
+        active_cache_dev = casadm.start_cache(active_cache_dev, cache_line_size=cache_line_size, force=True)
+        active_cache_dev.stop()
+
+    with TestRun.step("Create a symbolic link to the second partition."):
+        link_path = "/dev/disk/by-id/standby_cache_dev_link"
+        symlink = Symlink.create_symlink(link_path, standby_cache_dev.path, True)
+    with TestRun.step("Start a standby instance on the second partition using original path."):
+        standby_cache = casadm.standby_init(
+            cache_dev=standby_cache_dev,
+            cache_id=cache_id,
+            cache_line_size=cache_line_size,
+            force=True
+        )
+
+    with TestRun.step(
+        "Populate the standby cache with valid CAS metadata from the first partition."
+    ):
+        Dd().input(cache_dev.partitions[0].path).output(f"/dev/cas-cache-{cache_id}").run()
+
+    with TestRun.step("Detach standby cache."):
+        standby_cache.standby_detach()
+
+    with TestRun.step("Activate cache providing symlink."):
+        output = TestRun.executor.run(
+            standby_activate_cmd(cache_dev=symlink.get_symlink_path(), cache_id=str(cache_id))
+        )
+        if output.exit_code != 0:
+            TestRun.LOGGER.error("Failed to activate cache with provided symlink.")
+
+    with TestRun.step(
+        "Verify that the activation succeeded, and the regular cache instance is running."
+    ):
+        if standby_cache.get_status() != CacheStatus.running:
+            TestRun.LOGGER.error("Cache is not running.")
```

Comment:
maybe stop the cache and remove the symlink? our standard cleanup probably does not remove symbolic links created during the test.